src/ceph-volume/ceph_volume/util/disk.py

```diff
@@ -29,6 +29,55 @@ def get_partuuid(device):
     return ' '.join(out).strip()
 
 
+def udevadm_info(device):
+    """
+    Capture device information with udevadm, useful in situations where blkid and lsblk aren't
+    producing enough useful output.
+    """
+    command = ['udevadm', 'info', '--query=property', '--name', device]
+    out, err, rc = process.call(command)
+    return _udevadm_parser(out)
+
+
+def _udevadm_parser(output):
+    """
+    Parses the output from a udevadm call, requires output to be
+    produced using the ``info``, specifying the actual --query type, to remove prefixes in the output.
+
+    Expected output::
+
+        # udevadm info --query=property --name /dev/sdb
+        DEVLINKS=/dev/disk/by-path/pci-0000:00:14.0-scsi-0:0:3:0
+        DEVNAME=/dev/sdb
+        DEVPATH=/devices/pci0000:00/0000:00:14.0/host3/target3:0:3/3:0:3:0/block/sdb
+        DEVTYPE=disk
+        ID_BUS=scsi
+        ID_MODEL=HARDDISK
+        ID_MODEL_ENC=HARDDISK\x20\x20\x20\x20\x20\x20\x20\x20
+        ID_PART_TABLE_TYPE=gpt
+        ID_PART_TABLE_UUID=bea7a874-c406-46b2-98bd-414f062f2eb5
+        ID_PATH=pci-0000:00:14.0-scsi-0:0:3:0
+        ID_PATH_TAG=pci-0000_00_14_0-scsi-0_0_3_0
+        ID_REVISION=1.0
+        ID_SCSI=1
+        ID_TYPE=disk
+        ID_VENDOR=VBOX
+        ID_VENDOR_ENC=VBOX\x20\x20\x20\x20
+        MAJOR=8
+        ...
+    """
+    processed = {}
+    for line in output:
+        try:
+            key, value = line.split('=')
+        except ValueError:
```

Comment:
I don't have anything against this except for the silent error swallowing. I know there is a log statement now but the code still continues with bad input.

Reply:
Are you suggesting a hard fail? Or what would you propose here

Reply:
We could add a logging statement to the terminal. I would prefer not have a hard fail. An non-parseable line should not prevent gathering metadata for a partition

Reply:
I am going to respectfully disagree here. I understand your concern, but this is type of parsing and handling of output is how we process pretty much everything else in ceph-volume.

That doesn't mean it is a good reason to introduce something that can be improved, but this type of parsing is good enough for what we need, and IMO we shouldn't fail hard. In any case, the consumer of the produced output can chose to fail hard (as we do for `udevadm()`)

Reply:
Look if this scenario ever happens this code logs that there is something wrong with the output and then continue anyway with this unknown output (there might be a '=' in one of the output lines). I think this is not useful behaviour and potentially dangerous.

Reply:
The current usage is just to retrieve the partition number. I see your point of potential breakage with a `=` in the value side of things. 

That doesn't represent a dangerous behavior for determining partition numbers, but I agree that it might be better to have a strict mode where the caller may set a hard failure in case there are errors in parsing. 

I have created http://tracker.ceph.com/issues/37488 to follow up on that. The current use case does not warrant a strict mode.

Reply:
#25201 is ready to merge. We might as well just go with that.